source/MagicInternetBox.cpp

```diff
@@ -498,6 +501,10 @@ void MagicInternetBox::update(std::shared_ptr<ShipModel> state) {
 				break;
 		}
 
+		if (state->isLevelOver()) {
```

Comment:
Why does this happen after resolving a state sync? shouldn't this happen inside resolveState?

Reply:
This is needed here because the lines that follow are the ones that process gameplay messages, which we want to skip once the game has ended. I'll add another check inside state sync, although we've yet to see a crash from state sync, since state sync doesn't deal with buttons at the moment.